src/videojs.ads.js

```diff
@@ -617,19 +617,16 @@ var
             if (!videojs.browser.IS_IOS && player.duration() === Infinity) {
               player.volume(this.preAdVolume_);
             }
-
-            // trigger 'adend' as a consistent notification
-            // event that we're exiting ad-playback.
-            if (player.ads.triggerevent !== 'adend') {
-              player.trigger('adend');
```

Comment:
So, because 'adend' was triggered both in endLinearAdMode and in ad-playback.leave, that was why there were duplicate ad-ends? And you're removing this one and making an error-specific one. Just want to make sure I understand what's going on here.